### PR TITLE
Infoblox_nios: add client geoip and registered_domain processors to dns data

### PR DESCRIPTION
**Pull request by philippkahr**
When looking at the differences between the infoblox NIOS DNS dataset and our `network_packet_capture` dataset I discovered that we are not using `dns.flags` to print out the DNS flags such as: 

```
- = recursion not available
+ = recursion available (from DNS message header)
A = authoritative answer (from DNS message header)
t = truncated response (from DNS message header)
E = EDNS OPT record present (from DNS message header)
D = DNSSEC OK (from EDNS OPT RR)
V = responding server has validated DNSSEC records
L = response contains DTC synthetic record
```

Taken from: [Infoblox DNS](https://docs.infoblox.com/space/NAG8/22252254/Capturing+DNS+Queries+and+Responses)

This is an example taken from our `network packet capture dns` dataset:

```json
"dns": {
      "response_code": "NOERROR",
      "resolved_ip": [
        "34.107.117.83"
      ],
      "question": {
        "registered_domain": "es.io",
        "top_level_domain": "io",
        "etld_plus_one": "es.io",
        "name": "00d0ca5813b840dca7cee7ecebd65591.europe-west3.gcp.cloud.es.io",
        "subdomain": "00d0ca5813b840dca7cee7ecebd65591.europe-west3.gcp.cloud",
        "type": "A",
        "class": "IN"
      },
      "flags": {
        "authoritative": false,
        "truncated_response": false,
        "recursion_desired": true,
        "recursion_available": true,
        "checking_disabled": false,
        "authentic_data": false
      },
```

I added the following:

* ~couple of `set` processors to add the flags~
* registered domain processor to parse out the subdomain, top_level and so on.
* Added geoIP processing for `client.ip`.

I am not sure about the version bump, if this should become 1.17.0 or 2.0.0.

## Proposed commit message

This adds a `geoip` processor for the `client.ip` field and a `registered_domain` processor for the `dns.question.name` field.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/infoblox_nios/data_stream/log/fields/fields.yml [hunk 1]
```diff
@@ -139,3 +139,21 @@
       type: keyword
     - name: type
       type: keyword
+- name: dns
+  type: group
+  fields:
+    - name: flags
```
Comment (efd6) [inline packages/infoblox_nios/data_stream/log/fields/fields.yml:145]: I'm not sure we should be doing this.

@andrewkroh

Comment (philippkahr) [reply packages/infoblox_nios/data_stream/log/fields/fields.yml:145]: Not sure about it myself. We are creating fields in the ECS terrority. At the same time with the `dns.answers.ttl` object when containing multiple answers, we are already breaking ECS definition in another place.

According to ECS `dns.answers.ttl` is a `long`. [ecs](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html#field-dns-answers-ttl)

However, even in our sample logs we are already creating arrays, as depicted here: https://github.com/elastic/integrations/blob/cd7862ad9b612b1b38db80abb858192186e7fe8e/packages/infoblox_nios/data_stream/log/_dev/test/pipeline/test-dns.log-expected.json#L310

```
        "ttl": [
            28800,
            28800
        ],
```


Also the `network_traffic` integration for the `DNS` datastreams just bluntly writes it all into the `dns` object as well: https://github.com/elastic/integrations/blob/cd7862ad9b612b1b38db80abb858192186e7fe8e/packages/network_traffic/data_stream/dns/fields/protocol.yml#L1

Comment (efd6) [reply packages/infoblox_nios/data_stream/log/fields/fields.yml:145]: Looking at the [dns fields in ECS](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html), the flags here should be [`dns.header_flags`](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html#field-dns-header-flags). This solves the issue at least for this. The array/scalar issue for `dns.answers.ttl` is more difficult and something that we see in other fields as well. There are more egregious issues in network_traffic which we plan to address.

Comment (philippkahr) [reply packages/infoblox_nios/data_stream/log/fields/fields.yml:145]: I see, for me the `dns.header_flags` are hard to work with, as they are just abbreviations. That's why I like the network traffic approach of translating the flags to actual fields where I can then say `dns.flags.authoritive_answer: true`

Comment (efd6) [reply packages/infoblox_nios/data_stream/log/fields/fields.yml:145]: Unfortunately, the ECS is pretty clear on this.

**Comment by philippkahr** [comment]
@efd6  did remove the non ecs conforming fields. Does the rest look good?

**Comment by philippkahr** [comment]
The `dns.header_flags` are already set here: https://github.com/elastic/integrations/blob/cd0a5ec83f84a123770027b3289989b2b3fa679d/packages/infoblox_nios/data_stream/log/elasticsearch/ingest_pipeline/pipeline_dns.yml#L210

**Comment by efd6** [comment]
🤦 
Thanks. I will take a look in the morning.

**Review by efd6 — COMMENTED** [review]
Please update the proposed commit message to match the current state of the change.

**Comment by philippkahr** [comment]
Can you take care of it please? I am out on PTO for another full week :)

**Review by efd6 — APPROVED** [review]

**Pull request merged**